### PR TITLE
refactor: 💡 mirage changes to support ldap account type

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: ICU-10052

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10052)

## Description
Only added changes to mirage factories. Looks like no changes necessary in mirage serializer and config.

:technologist: [Admin preview](PATH_TO_FEATURE)

### Screenshots (if appropriate):
New Note

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]

File: addons/api/mirage/factories/account.js [hunk 1]
```diff
@@ -11,32 +11,48 @@ import {
...
-      'no-op',
-      'read',
-      'update',
-      'delete',
-      'set-password',
-    ],
+  authorized_actions() {
+    let authorizedActions = ['no-op', 'read', 'update', 'delete'];
+    if (this.type === TYPE_AUTH_METHOD_PASSWORD)
+      authorizedActions.push('set-password');
```
Comment (ZedLi) [inline addons/api/mirage/factories/account.js:21]: Could you add braces if the `if` statement isn't on one line?

File: addons/api/mirage/factories/account.js [hunk 2]
```diff
@@ -11,32 +11,48 @@ import {
...
-          subject: 'sub',
-          email: faker.internet.email(),
-          full_name: faker.random.words(),
-        };
+    if (this.type === TYPE_AUTH_METHOD_PASSWORD) {
+      return { login_name: faker.internet.userName() };
+    }
+
+    if (this.type === TYPE_AUTH_METHOD_LDAP) {
+      const groupsAmount = faker.datatype.number({ min: 1, max: 3 });
```
Comment (ZedLi) [inline addons/api/mirage/factories/account.js:35]: I'm curious, why did you change this from a `switch` to `if` statements? Is it because of the variable declaration in this block and the linter complained?

If so you can get around that by explicitly making a switch case statement have a block, i.e. wrap it in braces

**Review by ZedLi — APPROVED** [review]

File: addons/api/mirage/factories/account.js (hunk 2 above)
Comment (lisbet-alvarez) [reply addons/api/mirage/factories/account.js]: Yup, exactly the reason. Thank you! Updated!

**Review by DhariniJeeva — APPROVED** [review]
